tests/python/test_errors.py

```diff
@@ -740,6 +740,27 @@ def test_bad_type_get_dataset_list_range(use_cluster):
     with pytest.raises(TypeError):
         c.get_dataset_list_range(listname, start_index, "not an integer")
 
+def test_bad_type_log_data(use_cluster, context):
+    c = Client(None, use_cluster, logger_name=context)
+    with pytest.raises(TypeError):
+        log_data("Not a logging level", "Data to be logged")
+    with pytest.raises(TypeError):
+        log_data(LLInfo, 42)
+
+def test_bad_type_log_warning(use_cluster, context):
+    c = Client(None, use_cluster, logger_name=context)
+    with pytest.raises(TypeError):
+        log_warning("Not a logging level", "Data to be logged")
+    with pytest.raises(TypeError):
+        log_warning(LLInfo, 42)
+
+def test_bad_type_log_error(use_cluster, context):
```

Comment:
at risk of over complicating the test suite, you could do something like
```py
@pytest.mark.parameterize("log_fn", [
    (log_data,), (log_warning,), (log_error,)
])
def test_bad_log_level(use_cluster, context, log_fn):
    ...
```
so you that the same test doesn't have to get rewritten for any/every new log fns that may get made in future (assuming the all take a log level as the first param)

Reply:
good thought -- done